cpp/src/arrow/memory_pool.cc

```diff
@@ -44,7 +45,13 @@ namespace {
 // Allocate memory according to the alignment requirements for Arrow
 // (as of May 2016 64 bytes)
 Status AllocateAligned(int64_t size, uint8_t** out) {
-// TODO(emkornfield) find something compatible with windows
+  // TODO(emkornfield) find something compatible with windows
```

Comment:
I'm sceptical of this new check for performance reason. Maybe hide all of this under asserts such that it is avoided in production?

Reply:
Do you have benchmarks to demonstrate the performance loss here? If yes, how do I run them?

Reply:
I think those checks are reasonable. The Arrow allocator will usally be called with large-ish sizes (i.e. we're not using it to allocate tons of tiny objects), so the check shouldn't be problematic.